e2e/cypress/support/api/oidc-settings.ts

```diff
@@ -1,30 +1,32 @@
-import { apiCallProperties } from './apiauth';
-import { ensureSomethingIsSet } from './ensure';
+import { ensureSetting } from './ensure';
+import { API } from './types';
 
 export function ensureOIDCSettingsSet(
-  api: apiCallProperties,
-  accessTokenLifetime,
-  idTokenLifetime,
-  refreshTokenExpiration,
+  api: API,
+  accessTokenLifetime: number,
+  idTokenLifetime: number,
+  refreshTokenExpiration: number,
   refreshTokenIdleExpiration: number,
 ): Cypress.Chainable<number> {
-  return ensureSomethingIsSet(
+  return ensureSetting(
     api,
     `${api.adminBaseURL}settings/oidc`,
-    (settings: any) => {
-      let entity = null;
+    (body: any) => {
+      const result = {
+        entity: body.settings,
+        sequence: body.settings?.details?.sequence,
+        id: body.settings.id,
+      };
+
       if (
-        settings.settings?.accessTokenLifetime === hoursToDuration(accessTokenLifetime) &&
-        settings.settings?.idTokenLifetime === hoursToDuration(idTokenLifetime) &&
-        settings.settings?.refreshTokenExpiration === daysToDuration(refreshTokenExpiration) &&
-        settings.settings?.refreshTokenIdleExpiration === daysToDuration(refreshTokenIdleExpiration)
+        body.settings?.accessTokenLifetime != hoursToDuration(accessTokenLifetime) ||
+        body.settings?.idTokenLifetime != hoursToDuration(idTokenLifetime) ||
+        body.settings?.refreshTokenExpiration != daysToDuration(refreshTokenExpiration) ||
+        body.settings?.refreshTokenIdleExpiration != daysToDuration(refreshTokenIdleExpiration)
       ) {
-        entity = settings.settings;
+        result.entity = null;
```

Comment:
why not invert this? so that all equal === and then set the result, maybe use spread operator

Reply:
how does the spread operator help here?

I could return immediately with `{ entity: <typestrict checks> ? body.settings : null, sequence: body.settings?.details?.sequence, id: body.settings.id }`

Or what do you mean?

Reply:
or do you just mean this?

```
      const result = {
        sequence: body.settings?.details?.sequence,
        id: body.settings.id,
        entity: null,
      };

      if (
        body.settings?.accessTokenLifetime === hoursToDuration(accessTokenLifetime) ||
        body.settings?.idTokenLifetime === hoursToDuration(idTokenLifetime) ||
        body.settings?.refreshTokenExpiration === daysToDuration(refreshTokenExpiration) ||
        body.settings?.refreshTokenIdleExpiration === daysToDuration(refreshTokenIdleExpiration)
      ) {
        result.entity = body.settings;
      }
      return result;
```

Reply:
shouldn't the check be made with && so,
```
const result = {
      sequence: body.settings?.details?.sequence,
      id: body.settings.id,
    };
if (exp && exp && exp && exp) {
return { entity, ...result };
} else {
return result;
}
```

Reply:
entity must not be null, the typescript transpiler, complains :-)

Reply:
so adding body.settings to the if above helps?

Reply:
like this? https://github.com/zitadel/zitadel/pull/4342/commits/164fdc6010c09c5326eac353728db730aef71f07

Reply:
meant more like this, should the expressions not be concatenated with && 

```
 if (
        body.settings &&
        body.settings.accessTokenLifetime === hoursToDuration(accessTokenLifetime) &&
        body.settings.idTokenLifetime === hoursToDuration(idTokenLifetime) &&
        body.settings.refreshTokenExpiration === daysToDuration(refreshTokenExpiration) &&
        body.settings.refreshTokenIdleExpiration === daysToDuration(refreshTokenIdleExpiration)
      ) {
        return { entity: body.settings, ...result };
      }
```